src/ResourceManager/Sql/Commands.Sql/Instance Failover Group/Cmdlet/NewAzureSqlInstanceFailoverGroup.cs

```diff
@@ -0,0 +1,189 @@
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The name of the secondary resource group of the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the partner region for Azure SQL Database Instance Failover Group
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the secondary server of the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerRegion { get; set; }
+        
+        /// <summary>
+        /// Gets or sets the name of the Managed Instance on the local region
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the Managed Instance on the local region to be added to the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PrimaryManagedInstanceName { get; set; }
+        
+        /// <summary>
+        /// Gets or sets the name of the Managed Instance on the partner region
+        /// </summary>
+        [Parameter(Mandatory = true,
+            HelpMessage = "The name of the Managed Instance on the partner region to be added to the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        public string PartnerManagedInstanceName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy without data loss for the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "The failover policy of the Azure SQL Database Instance Failover Group.")]
+        [ValidateNotNullOrEmpty]
+        [PSDefaultValue(Help = "Automatic")]
+        public FailoverPolicy FailoverPolicy { get; set; }
+
+        /// <summary>
+        /// Gets or sets the grace period with data loss for the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Interval before automatic failover is initiated if an outage occurs on the primary server and failover cannot be completed without data loss.")]
+        [ValidateNotNullOrEmpty]
+        [ValidateRange(1, int.MaxValue/60)]
+        [PSDefaultValue(Help = "1")]
+        public int GracePeriodWithDataLossHours { get; set; }
+
+        /// <summary>
+        /// Gets or sets the failover policy for read only endpoint of the Sql Azure Instance Failover Group.
+        /// </summary>
+        [Parameter(Mandatory = false,
+            HelpMessage = "Whether an outage on the secondary server should trigger automatic failover of the read-only endpoint. This feature is not yet supported.")]
+        [ValidateNotNullOrEmpty]
+        public AllowReadOnlyFailoverToPrimary AllowReadOnlyFailoverToPrimary { get; set; }
```

Comment:
Can this be SwitchParameter instead?

Reply:
We kept it like this so it stays the same as the current Failover Groups.